### PR TITLE
[SYCLomatic][CMake] Fix bug issue that library cuda is  not removed when it is used in CMake list command

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml [hunk 1]
```diff
@@ -2258,3 +2258,17 @@
...
+  MatchMode: Partial
+  CmakeSyntax: lib_cuda
+  In: list${empty}(${value})
+  Out: list${empty}(${value})
+  Subrules:
+    value:
+      MatchMode: Full
+      In: cuda
+      Out: ""
+      RuleId: "remove_lib_cuda"
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml:2274]:
```suggestion
      RuleId: "remove_lib_cuda"
      warning:  "Warning DPCT30001 please revisit the migrate scripts"
```

Comment (zhimingwang36) [reply clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml:2274]: or you may follow up in a separate PR.

File: clang/test/dpct/cmake_migration/case_020/input.cmake [hunk 2]
```diff
@@ -1,3 +1,4 @@
 set(SOURCES foo.cu ${CMAKE_SOURCE_DIR}/foo/bar.cu)
 list(APPEND SOURCES foo.dp.cpp)
 list(append my_libs ${CUDA_LIBRARIES})
+list(APPEND FOO_LIBRARIES cuda)
```
Comment (zhimingwang36) [inline clang/test/dpct/cmake_migration/case_020/input.cmake:4]: wht if list(Append  file_list  cuda)?  suggest to emit msg.

**Review by zhimingwang36 — APPROVED** [review]
Merge this PR 1st to unblock the test. Expect further PR to refine the rule: rule_lib_cuda_removed.

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml (hunk 1 above)
Comment (tomflinda) [reply clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml:2274]: Will implement it in a separate PR.

**Pull request merged**